stdlib/LinearAlgebra/src/blas.jl

```diff
@@ -799,6 +799,9 @@ for (fname, elty, lib) in ((:dsymv_,:Float64,libblastrampoline),
         function symv!(uplo::AbstractChar, alpha::Union{($elty), Bool},
                        A::AbstractMatrix{$elty}, x::AbstractVector{$elty},
                        beta::Union{($elty), Bool}, y::AbstractVector{$elty})
+            if uplo ∉ ('U', 'L')
```

Comment:
Could be factored out to a function. (I believe such a uplo-check function exist somewhere already?)

Reply:
Not sure if there is, but good to use it. A compact code pattern:

```
uplo ∉ ('U', 'L') && throw...
```

Reply:
https://github.com/JuliaLang/julia/blob/d44a53425ba9cb320408e71206d600513dd2ac9e/stdlib/LinearAlgebra/src/lapack.jl#L60

Reply:
Is that available outside the `LAPACK`-submodule? Or do I need to cross-import it somehow into the `BLAS` submodule?

Reply:
Hm, the BLAS module can't import something from the LAPACK module because the latter is loaded later. That `chkuplo` function is embedded in several other LAPACK-style check functions. So what is "best practices" here? Just copy-paste the same function and keep things defined locally? Move that one function from LAPACK to BLAS and import from LAPACK?

Reply:
Move the function out of both maybe?

Reply:
This is just a detail anyway, but I moved the function over to BLAS, and use it in LAPACK now. Moving out of both could be potentially confusing since we don't use characters but symbols in some of the `*diagonal` code for uplo's. There, we have similar input checks, but not factored out as a function. Maybe in another step we take another fresh look if we want to hoist it out further and make things more consistent. For now, BLAS and LAPACK are consistent.